src/conversations/conversations.gateway.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common'
+import { AuthService } from '../auth/auth.service'
+ 
+import { WebSocketGateway, OnGatewayConnection, WebSocketServer } from '@nestjs/websockets'
+
+@Injectable()
+@WebSocketGateway()
+export class ConversationsGateway implements OnGatewayConnection {
+  constructor(private readonly authService: AuthService) {}
+  @WebSocketServer() server;
+
+  async handleConnection(socket) {
+    try {
+      await this.authService.validateJWT(socket.handshake.headers.token)
+    } catch (err) {
+      socket.disconnect(true)
```

Comment:
is there a way to send an error code/message on disconnect? Otherwise you will never know why you got kicked out

Reply:
Nope, what I could do is send a message with the reason first and then disconnect...
Not really sure how I feel about that though...

Reply:
Hmm, That could have some security issues